htdocs/core/lib/modulebuilder.lib.php

```diff
@@ -663,8 +663,7 @@ function writePropsInAsciiDoc($file, $objectname, $destfile)
 		$string = trim($string, "'");
 		$string = rtrim($string, ",");
 
-		$array = [];
-		eval("\$array = [$string];");
+		$array = eval("return [$string];");
```

Comment:
Try to use dol_eval()

Reply:
with do_eval it's not the result I want, but I've taken the necessary steps to work with eval by cleaning the string and checking if it's an array

Reply:
Try 
dol_eval("return $string;", 1);
and better
dol_eval("$string");

Reply:
do_eval not work for me so i found other way to do it